cairo-libs/packages/cometbft/src/types.cairo

```diff
@@ -309,33 +309,62 @@ impl SignedHeaderAsProtoName of ProtoName<SignedHeader> {
     }
 }
 
-#[derive(Default, Debug, Clone, Drop, PartialEq, Serde)]
+#[derive(Drop, Default, Debug, Clone, PartialEq, Serde)]
 pub struct PublicKey {
-    // TODO(rano): this is oneof
-    pub ed25519: ByteArray,
-    pub secp256k1: ByteArray,
+    pub sum: Sum,
 }
 
 impl PublicKeyAsProtoMessage of ProtoMessage<PublicKey> {
     fn encode_raw(self: @PublicKey, ref context: EncodeContext) {
-        context.encode_field(1, self.ed25519);
-        context.encode_field(2, self.secp256k1);
+        context.encode_oneof(self.sum)
     }
 
     fn decode_raw(ref context: DecodeContext) -> Option<PublicKey> {
-        let ed25519 = context.decode_field(1)?;
-        let secp256k1 = context.decode_field(2)?;
-        Option::Some(PublicKey { ed25519, secp256k1 })
+        let sum = context.decode_oneof()?;
+        Option::Some(PublicKey { sum })
     }
 
     fn wire_type() -> WireType {
         WireType::LengthDelimited
     }
 }
 
-impl PublicKeyAsProtoName of ProtoName<PublicKey> {
-    fn type_url() -> ByteArray {
-        "PublicKey"
+#[derive(Drop, Default, Debug, Clone, PartialEq, Serde)]
+pub enum Sum {
+    #[default]
+    Ed25519: Array<u8>,
```

Comment:
Moving towards `Array<u8>,` since in a later step, we’ll need to work with bytes for the signature verification.